### PR TITLE
[Ready] Basic Driver

### PR DESCRIPTION
**Pull request by revett**
### Problem / Solution

I can't increment a custom Statsd metric using Alephant Logger.

~~**Note** - had to add a 1 second sleep within 2 tests as there was a race condition where the Thread didn't spin up in time.~~

### CODE REVIEW

File: spec/statsd_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        sleep 1
+        expect(server).to have_received(:increment).once.with(key, 1)
+      end
+    end
+
+    context "custom interval" do
+      let(:interval) { 10 }
+
+      specify do
+        driver.increment(key, interval)
```
Comment (darnould) [inline spec/statsd_spec.rb]: Whack `.join` on the end of here and don't sleep

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+language: ruby
+rvm:
+  - "jruby"
```
Comment (Integralist) [inline .travis.yml]: You don't need the quotes around the value. Also, do we not want to specify a version?

e.g.

``` yaml
rvm:
  - jruby
  - 1.7.17
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+script:
+  - bundle exec rake all
```
Comment (Integralist) [inline .travis.yml]: I think with a  one line script you can just use:

``` yaml
script: "bundle exec rake all"
```

...again, you might not need the quotes there

File: spec/statsd_spec.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        expect(server).to have_received(:namespace=).once.with(
+          config[:namespace]
+        )
+      end
+    end 
+  end
+
+  describe "#increment" do
+    let(:driver) { subject.new }
+    let(:key) { 'batman' }
```
Comment (Integralist) [inline spec/statsd_spec.rb]: Letting the single quotes slip in still ;-)

**Comment by Integralist** [comment]
@revett this all looks fine. The only concern is how can we avoid the Thread's blocking nature in the test (e.g. we have to wait for the thread to finish, meaning a potentially slower test).

**Comment by darnould** [comment]
@Integralist It shouldn't be any slower than if it _wasn't_ in a thread - unless the OS decided to wait years to schedule it: which none would do in this day & age :)

File: lib/alephant/logger/statsd.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      def increment(key, interval = 1)
+        send { server.increment(key, interval) }
+      end
+
+      private
+
+      attr_reader :server
+
+      def connect(config)
+        @server = ::Statsd.new(config[:host], config[:port]).tap do |s|
```
Comment (darnould) [inline lib/alephant/logger/statsd.rb]: Move assignment to ctor as discussed

**Comment by darnould** [comment]
Glorious :cake: after comments

**Pull request closed without merging**